src/Scripting/CoreTestUtilities/RedirectedOutput.cs

```diff
@@ -2,16 +2,21 @@
 
 using System;
 using System.IO;
+using System.Threading;
 
 namespace Microsoft.CodeAnalysis.Scripting
 {
     public sealed class OutputRedirect : IDisposable
     {
+        private static readonly object s_guard = new object();
+
         private readonly TextWriter _oldOut;
         private readonly StringWriter _newOut;
 
         public OutputRedirect(IFormatProvider formatProvider)
         {
+            Monitor.Enter(s_guard);
```

Comment:
Using a TestCollection to guarantee the tests in question (and only those tests) are serialized wrt everything else.